scene/gui/text_edit.cpp

```diff
@@ -1627,52 +1686,36 @@ void TextEdit::_consume_pair_symbol(char32_t ch) {
 		}
 	}
 
-	String line = text[cursor.line];
-
-	bool in_single_quote = false;
-	bool in_double_quote = false;
-	bool found_comment = false;
-
-	int c = 0;
-	while (c < line.length()) {
-		if (line[c] == '\\') {
-			c++; // Skip quoted anything.
-
-			if (cursor.column == c) {
-				break;
-			}
-		} else if (!in_single_quote && !in_double_quote && line[c] == '#') {
-			found_comment = true;
-			break;
-		} else {
-			if (line[c] == '\'' && !in_double_quote) {
-				in_single_quote = !in_single_quote;
-			} else if (line[c] == '"' && !in_single_quote) {
-				in_double_quote = !in_double_quote;
-			}
-		}
-
-		c++;
-
-		if (cursor.column == c) {
-			break;
-		}
-	}
-
-	// Do not need to duplicate quotes while in comments
-	if (found_comment) {
+	// Disallow inserting duplicate quotes while inside quotes or comments
+	if (_get_running_quotes(line_no, col_no) != "") {
 		insert_text_at_cursor(ch_single);
 		cursor_set_column(cursor_position_to_move);
 
 		return;
 	}
 
-	// Disallow inserting duplicated quotes while already in string
-	if ((in_single_quote || in_double_quote) && (ch == '"' || ch == '\'')) {
-		insert_text_at_cursor(ch_single);
-		cursor_set_column(cursor_position_to_move);
+	//detect if its the case of """ <text> """ ie. multiline string...
+	//otherwise dont change the initial values of ch_single, ch_single_pair and ch_pair
```

Comment:
```suggestion
	// Detect if it's the case of """ <text> """ i.e. multiline
	// string, otherwise don't change the initial values of
	// `ch_single`, `ch_single_pair` and `ch_pair`.
```